### PR TITLE
USHIFT-286: Run Service CA component as non root

### PR DESCRIPTION
**Pull request by oglok**
Signed-off-by: Ricardo Noriega <rnoriega@redhat.com>

**Which issue(s) this PR addresses**:

Closes [USHIFT-286](https://issues.redhat.com/browse/USHIFT-286)

### CODE REVIEW

File: assets/components/service-ca/deployment.yaml [hunk 1]
```diff
@@ -23,7 +23,10 @@ spec:
         app: service-ca
         service-ca: "true"
     spec:
-      securityContext: {}
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 1001
```
Comment (dhellmann) [inline assets/components/service-ca/deployment.yaml:28]: Is there any reason to register or document this user ID to ensure that other workloads do not reuse it?

**Review by dhellmann — COMMENTED** [review]
One question. If the answer is "no", feel free to remove the hold and merge this as it is.

/hold
/lgtm

File: assets/components/service-ca/deployment.yaml (hunk 1 above)
Comment (dhellmann) [reply assets/components/service-ca/deployment.yaml:28]: OK, I don't know how UIDs like this are applied. If we have another pod with the same settings, would it be able to see or communicate with this one in a way we wouldn't want?

Comment (oglok) [reply assets/components/service-ca/deployment.yaml]: This isn't right. I've override the existing commit. The Cluster Policy Controller should generate the UID based on the security context policy, in this case, `runAsNonRoot`. 

After fixing the cluster role binding, now it works:

```
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsNonRoot: true
      runAsUser: 1000070000
      ```

**Pull request merged**